### PR TITLE
feat: accept Polymarket URLs as input

### PR DESCRIPTION
**Pull request by guneysol**
## Summary
- `markets get` and `events get` now accept full Polymarket URLs in addition to numeric IDs and slugs
- Extracts the correct slug automatically (market slug for `markets get`, event slug for `events get`)
- Handles scheme variations, `www.`, query strings, fragments, and trailing slashes

## Motivation
Agents and scripts commonly receive Polymarket URLs from users. This removes the need to manually extract slugs before passing them to the CLI.

## Test plan
- 22 unit tests covering URL parsing and `resolve_id` logic
- 2 integration tests verifying URL acceptance
- Backward compatible — plain slugs and numeric IDs work as before

---

> [!NOTE]
> **Low Risk**
> Small, isolated CLI input-parsing change with thorough unit/integration test coverage; main risk is edge-case misparsing leading to fetching the wrong slug.
> 
> **Overview**
> `markets get` and `events get` now accept full Polymarket URLs in addition to numeric IDs and slugs by introducing `parse_polymarket_url`/`resolve_id` and routing lookups based on the resolved `ResolvedId`.
> 
> URL handling normalizes scheme/`www`, strips query/fragment/trailing slashes, and picks the **market slug** for `markets get` (when present) vs the **event slug** for `events get`; coverage is added via new unit tests in `commands/mod.rs` plus integration tests ensuring URLs aren’t rejected at CLI parsing.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit b977baa305e876de80bbe176ccb40b0c1ab618ac. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/commands/mod.rs [hunk 1]
```diff
@@ -30,10 +30,106 @@ pub fn parse_condition_id(s: &str) -> anyhow::Result<B256> {
...
+/// slug is used; otherwise the event slug is used.
+pub fn resolve_id(input: &str, prefer_market: bool) -> ResolvedId {
+    if let Some(parsed) = parse_polymarket_url(input) {
+        let slug = if prefer_market {
+            parsed.market_slug.unwrap_or(parsed.event_slug)
+        } else {
+            parsed.event_slug
+        };
+        return ResolvedId::Slug(slug);
+    }
```
[the remark this answers is not in the document]
Comment (guneysol) [reply src/commands/mod.rs:118]: Intentional. When the URL has no market slug, falling back to the event slug is better than erroring -- it gives the API a chance to resolve it (works when event/market slugs match, which is common for single-market events). 

When it doesn't match, the API returns a clear 404 ("slug not found"), same as typing any wrong slug. No silent misresolution is possible.
